### ISSUE TITLE
[BUG] Route editor bug

### ISSUE THREAD

**Issue by sgtoutlaw** [EliteMasterEric/Teyvat.moe#147]
**Describe the bug**
The page blanks out when trying to put anything into the title or description fields of route edits.

**To Reproduce**
1. Go to the map
2. Enable the editor mode.
3. Use the routing function to make a route.
4. Navigate to the elements tab and try to type or paste anything into the Title or Description Fields.
5. The page will immediately leave you with only a blank background.

**Expected behavior**
I expected to be able to enter a title or description.  Even trying to write it elsewhere and paste it caused the same problem.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser firefox, chrome
 - Version 84.0 64 bit, 87.0.4280.88 64-bit)

**Additional context**
The field that is for putting a video or image url doesn't seem to have the same problem.

https://user-images.githubusercontent.com/35310360/102703562-4bad8280-423e-11eb-960d-7b05766a6a7b.mp4

**Comment by EliteMasterEric** [comment]
Hey!

Thanks for taking the time to write a detailed issue report.

The app appears to be crashing here, and I haven't been able to reproduce the issue.

Can you try opening the browser console (press F12 in Firefox or Chrome) and see if an error appears there when performing these steps? I might be able to diagnose the issue based on that.

**Comment by sgtoutlaw** [comment]
`Content Security Policy: The page’s settings blocked the loading of a resource at blob:https://genshinmap.github.io/ddb0e13e-4e10-44e6-aad5-bce3a9e40994 (“img-src”). leaflet.textpath.js:88:34
COULD NOT HASH OBJECT Util.js:57:12
    je Util.js:57
    up EditorLayer.js:104
    React 8
    unstable_runWithPriority scheduler.production.min.js:18
    React 10
    unstable_runWithPriority scheduler.production.min.js:18
    React 11
    <anonymous> index.js:13
    i (index):1
    t (index):1
    r (index):1
    <anonymous> main.3cfb1c23.chunk.js:1
{…}
​
data: Array [ {…} ]
​
highlighted: -1
​
icons: Object { base: {…}, done: {…} }
​
<prototype>: Object { … }
Util.js:58:12
    je Util.js:58
    up EditorLayer.js:104
    React 8
    unstable_runWithPriority scheduler.production.min.js:18
    React 10
    unstable_runWithPriority scheduler.production.min.js:18
    React 11
    <anonymous> index.js:13
    i (index):1
    t (index):1
    r (index):1
    <anonymous> main.3cfb1c23.chunk.js:1
Error: Unknown object type "inputevent"
    _object object_hash.js:1
    dispatch object_hash.js:1
    _object object_hash.js:1
    _object object_hash.js:1
    dispatch object_hash.js:1
    _object object_hash.js:1
    _object object_hash.js:1
    dispatch object_hash.js:1
    _object object_hash.js:1
    _object object_hash.js:1
    dispatch object_hash.js:1
    _object object_hash.js:1
    _object object_hash.js:1
    dispatch object_hash.js:1
    _array object_hash.js:1
    _array object_hash.js:1
    _object object_hash.js:1
    dispatch object_hash.js:1
    _object object_hash.js:1
    _object object_hash.js:1
    dispatch object_hash.js:1
    h object_hash.js:1
    h object_hash.js:1
    je Util.js:55
    up EditorLayer.js:104
    React 8
    unstable_runWithPriority scheduler.production.min.js:18
    React 10
    unstable_runWithPriority scheduler.production.min.js:18
    React 11
    <anonymous> index.js:13
    i (index):1
    t (index):1
    r (index):1
    <anonymous> main.3cfb1c23.chunk.js:1
react-dom.production.min.js:216:199
    React 5
    unstable_runWithPriority scheduler.production.min.js:18
    React 4
    unstable_runWithPriority scheduler.production.min.js:18
    React 10
    unstable_runWithPriority scheduler.production.min.js:18
    React 11
    <anonymous> index.js:13
    i (index):1
    t (index):1
    r (index):1
    <anonymous> main.3cfb1c23.chunk.js:1
Uncaught Error: Unknown object type "inputevent"
    _object object_hash.js:1
    dispatch object_hash.js:1
    _object object_hash.js:1
    _object object_hash.js:1
    dispatch object_hash.js:1
    _object object_hash.js:1
    _object object_hash.js:1
    dispatch object_hash.js:1
    _object object_hash.js:1
    _object object_hash.js:1
    dispatch object_hash.js:1
    _object object_hash.js:1
    _object object_hash.js:1
    dispatch object_hash.js:1
    _array object_hash.js:1
    _array object_hash.js:1
    _object object_hash.js:1
    dispatch object_hash.js:1
    _object object_hash.js:1
    _object object_hash.js:1
    dispatch object_hash.js:1
    h object_hash.js:1
    h object_hash.js:1
    je Util.js:55
    up EditorLayer.js:104
    React 8
    unstable_runWithPriority scheduler.production.min.js:18
    React 10
    unstable_runWithPriority scheduler.production.min.js:18
    React 11
    <anonymous> index.js:13
    i (index):1
    t (index):1
    r (index):1
    <anonymous> main.3cfb1c23.chunk.js:1
object_hash.js:1:3591
    _object object_hash.js:1
    dispatch object_hash.js:1
    _object object_hash.js:1
    forEach self-hosted:206
    _object object_hash.js:1
    dispatch object_hash.js:1
    _object object_hash.js:1
    forEach self-hosted:206
    _object object_hash.js:1
    dispatch object_hash.js:1
    _object object_hash.js:1
    forEach self-hosted:206
    _object object_hash.js:1
    dispatch object_hash.js:1
    _object object_hash.js:1
    forEach self-hosted:206
    _object object_hash.js:1
    dispatch object_hash.js:1
    _array object_hash.js:1
    forEach self-hosted:206
    _array object_hash.js:1
    _object object_hash.js:1
    dispatch object_hash.js:1
    _object object_hash.js:1
    forEach self-hosted:206
    _object object_hash.js:1
    dispatch object_hash.js:1
    h object_hash.js:1
    h object_hash.js:1
    je Util.js:55
    up EditorLayer.js:104
    React 7
    gc self-hosted:1208
    qn React
    unstable_runWithPriority scheduler.production.min.js:18
    React 10
    bind_applyFunctionN self-hosted:1359
    Xt self-hosted:1322
    unstable_runWithPriority scheduler.production.min.js:18
    React 3
    bind_applyFunctionN self-hosted:1359
    Yt self-hosted:1322
    React 3
    forEach self-hosted:4243
    React 5
    <anonymous> index.js:13
    i (index):1
    t (index):1
    r (index):1
    <anonymous> main.3cfb1c23.chunk.js:1
Content Security Policy: The page’s settings blocked the loading of a resource at inline (“script-src”). 53 fingerprint-protection.js:343:25
`

Sorry if that doesn't paste out well, but that is what the console says.  

Edit:  Wow, that looks even worse then I thought it was going to, let me take a couple pictures.

![firefox_QtD7kon3se](https://user-images.githubusercontent.com/35310360/102775829-a38ddb80-435b-11eb-823f-5c07006ba9cd.png)
![firefox_5JyCAbJODs](https://user-images.githubusercontent.com/35310360/102775750-74776a00-435b-11eb-9fda-69e2f1c2336e.png)

**Comment by EliteMasterEric** [comment]
Okay, so this error wass causing a page crash. The crash appears to be one I occasionally see, which is caused when the input of the text box is not being processed properly.

The reason this causes the page to display a blank screen is because the default behavior (upon experiencing an unhandled crash) is to drop the page completely. This is because I hadn't implemented a proper error handler.

Since this error occurred, the site has undergone a lot of improvements. I have reworked and tested that part of the user interface, so the bug should be resolved. Additionally, an error handler has been implemented, which should allow you to report this issue again easily if it happens in the future.
